feature/mypage/src/main/java/com/example/mypage/MyPageViewModel.kt

```
@@ -120,14 +153,25 @@ class MyPageViewModel @Inject constructor(
-                try {
-                    userRepository.logout() // 서버 로그아웃 + 토큰/유저ID 정리까지 Repository에서 처리
-                    onSuccess()
-                } catch (e: Exception) {
-                    onError("로그아웃에 실패했습니다.")
-                }
+    fun logout(onSuccess: () -> Unit, onError: (String) -> Unit) {
+        viewModelScope.launch {
+            try {
+                userRepository.logout()
+                onSuccess()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)